dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -62,10 +65,19 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
                 return;
             }
 
-            request.setRequestUri(uri);
             final PageMode mode = PageMode.getWithNavigateMode(request);
             try {
-                VelocityModeHandler.modeHandler(mode, request, response).serve();
+                final String pageHtml = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                        PageContextBuilder.builder()
+                                .setPageUri(uri)
+                                .setPageMode(mode)
+                                .setUser(APILocator.getLoginServiceAPI().getLoggedInUser())
+                                .build(),
+                        request,
+                        response
+                );
+
+                response.getOutputStream().write(pageHtml.getBytes());
```

Comment:
here I am using the API to render a Page this internally do a "response.getOutputStream().write(pageHtml.getBytes())", but doing first several check as the vanity URL